provider/oracle/network/environ.go

```diff
@@ -152,8 +155,12 @@ func (e Environ) getSubnetInfo() ([]network.SubnetInfo, error) {
 
 // NetworkInterfaces is defined on the environs.Networking interface.
 func (e Environ) NetworkInterfaces(instId instance.Id) ([]network.InterfaceInfo, error) {
-	id := string(instId)
-	instance, err := e.client.InstanceDetails(id)
+	providerId, err := e.env.ProviderID(instId)
```

Comment:
this feels a bit awkward: this will get all the instances to the ID, and then query the instance details again. why not just use the env.Instances method?

```go
instances, err := e.env.Instances([]instance.Id{instId})
if err != nil {
    return nil, errors.Trace(err)
}
instance := instances[0]
```

then you can drop the OracleInstancer interface, and just accept an environs.Environ in network.NewEnviron